### PR TITLE
Add support for client reset

### PR DESCRIPTION
**Pull request by leemaguire**
* Add support for the following client reset modes:
  - `realm::client_reset::manual()`
  - `realm::client_reset::discard_unsynced_changes(std::function<void(experimental::db local)> before, 
                                                   std::function<void(experimental::db local, experimental::db remote)> after)`
  - `realm::client_reset::recover_unsynced_changes(std::function<void(experimental::db local)> before,
                                                   std::function<void(experimental::db local, experimental::db remote)> after)`
  - `realm::client_reset::recover_or_discard_unsynced_changes(std::function<void(experimental::db local)> before,
                                                              std::function<void(experimental::db local, experimental::db remote)> after)`

### CODE REVIEW

File: tests/experimental/sync/client_reset_tests.cpp [hunk 1]
```diff
@@ -0,0 +1,372 @@
...
+                synced_realm.add(std::move(o));
+            });
+        }
+
+        auto synced_realm2 = experimental::db(flx_sync_config);
+        synced_realm2.refresh();
+
+        CHECK(error_handler_future.wait_for(std::chrono::milliseconds(15000)) == std::future_status::ready);
+    }
+}
```
Comment (jsflax) [inline tests/experimental/sync/client_reset_tests.cpp:372]: Add newline.

File: src/cpprealm/experimental/client_reset.cpp [hunk 2]
```diff
@@ -0,0 +1,36 @@
...
+        recover_unsynced_changes::recover_unsynced_changes(std::function<void(experimental::db local)> before,
+                                                           std::function<void(experimental::db local, experimental::db remote)> after) {
+            m_before = std::move(before);
+            m_after = std::move(after);
+            m_mode = internal::bridge::realm::client_reset_mode::recover;
+        }
+
+        recover_or_discard_unsynced_changes::recover_or_discard_unsynced_changes(std::function<void(experimental::db local)> before,
+                                                                                 std::function<void(experimental::db local, experimental::db remote)> after) {
+            m_before = std::move(before);
```
Comment (jsflax) [inline src/cpprealm/experimental/client_reset.cpp:31]: What is the point of these std::moves?

**Review by jsflax — APPROVED** [review]
Couple nits, otherwise lgtm.

File: src/cpprealm/experimental/client_reset.cpp (hunk 2 above)
Comment (leemaguire) [reply src/cpprealm/experimental/client_reset.cpp:31]: to avoid a copy and transfer ownership

**Pull request merged**